Add support for new OpenStack modules

emonty opened the pull request:
New module_utils functions, and a new inventory plugin needed by ansible/ansible-modules-extras#92

emonty committed:
Add support for new OpenStack modules

Incoming cloud config for OpenStack is complex due to plugins and
deployer choices. Rather than having the logic spread all over the
OpenStack modules, centralize it in the module_utils code.

emonty committed:
Add doc fragment for new OpenStack modules

jimi-c commented:
@emonty scratch that, as I realized you guys only have access to commit on the module repos.

emonty committed:
Add new OpenStack Inventory module

This inventory module is based on the shade library like the new os_
modules. It shares the ability to configure itself from os-client-config
configuration files or from the standard OS_ environment variables.

More importantly the guts of the code to get the server vars is now
shared with os_compute_facts. This means that playbooks that provision
compute hosts and then want to run plays on them can refer to server
variables in exactly the same way whether the provsioning play ran first
or whether the play ran in the context of a pre-existing inventory.
